agreement_sale_subscription_suspension/models/sale_subscription.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+
+
+class SaleSubscriptionSuspension(models.Model):
+    _inherit = 'sale.subscription'
+
+    def suspend_service_profile(self):
+        sp_ids = self.env['agreement.serviceprofile'].search([
+            ('agreement_id', '=', self.agreement_id.id)])
+        for sp_id in sp_ids:
+            sp_id.write({'state': 'suspended'})
+
+    def activate_service_profile(self):
+        sp_ids = self.env['agreement.serviceprofile'].search([
+            ('agreement_id', '=', self.agreement_id.id)])
+        for sp_id in sp_ids:
+            sp_id.write({'state': 'active'})
```

Comment:
```suggestion
            sp_ids.write({'state': 'active'})
```

Reply:
Remove
```
for sp_id in sp_ids:
            sp_id.write({'state': 'active'})
```
and do `sp_ids.write({'state': 'active'})`

write can be called in batch

Reply:
@Swapnesh-SerpentCS Thanks for your review. Once my commit suggestion will be add by @kvedder-amplex . It will take care everything. Afterwards, you may review it again.

Reply:
Oh, I reviewed this too fast. 
Sorry for this Noice. :+1:

Reply:
I stuck with the if statement for now. Let me know if more changes are requested.